src/cluster.c

```diff
@@ -656,7 +656,7 @@ void clusterReset(int hard) {
     if (nodeIsSlave(myself)) {
         clusterSetNodeAsMaster(myself);
         replicationUnsetMaster();
-        emptyDb(-1,EMPTYDB_NO_FLAGS,NULL);
+        emptyDb(-1,EMPTYDB_WITHFUNCTIONS,NULL);
```

Comment:
i think we may want to negate this flag.
so that by default it'll flush the functions, and only certain places will pass a flag to avoid that.

(initially when we discussed this, i thought we're adding another mandatory argument, which will fail compilation if someone forgot to set it to either 0 or 1)

Reply:
Fixed and update the top comment.